src/math.F90

```diff
@@ -659,4 +675,90 @@ function watt_spectrum(a, b) result(E_out)
 
   end function watt_spectrum
 
+!===============================================================================
+! W acts as a front end to the MIT Faddeeva function, Faddeeva_w.
+!===============================================================================
+
+  subroutine w(Z,wv)
+    use ISO_C_BINDING
+    complex(C_DOUBLE_COMPLEX), intent(inout) :: Z ! The point to evaluate Z at
+    complex(8), intent(out) :: wv                 ! The resulting w(Z) value
+    real(C_DOUBLE) :: relerr ! Target relative error in inner loop of MIT Faddeeva
+    logical        :: mangle ! Do we need to perform the special mangling step?
+
+    ! There is some special mangling the true W function in WHOPPER does.  I
+    ! replicate it here.
+    if(aimag(Z) < 0) then
+      Z = conjg(Z)
```

Comment:
Seems weird that this procedure actually changes the value of `Z`, which would be undesirable from the caller's perspective. Why not copy Z and take the conjugate on the local variable if need be? That would allow you to make the argument `intent(in)` rather than `intent(inout)`.